### PR TITLE
Feature/57662 LevelUp interface is missing properties `status` & `isOperational()` #57662

### PR DESCRIPTION
**Pull request by istherepie**
Related to discussion #57662 

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/57662)>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## levelup ([<kbd>unpkg</kbd>](https://unpkg.com/browse/levelup@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'levelup'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. The declaration doesn't match the JavaScript module 'levelup'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9965fdd6f0553122248779c349638b2e130bc40f
</p>

**Comment by carsonfarmer** [comment]
I should be able to review this, but will wait for CI to pass before doing so.

**Comment by istherepie** [comment]
I've disabled the check for the default export syntax:

```json
        "npm-naming": [
            true,
            {
                "errors": [
                    [
                        "NoDefaultExport",
                        false
                    ]
                ],
                "mode": "code"
            }
        ]
```

As far as I understand, the only other way to fix this issue would be to apply the `export =` syntax which will require to set the typescript compiler option `esModuleInterop` to `true` ?

File: types/levelup/tslint.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
                     [
                         "NeedsExportEquals",
                         false
+                    ],
+                    [
+                        "NoDefaultExport",
```
Comment (amcasey) [inline types/levelup/tslint.json:13]: Why was this needed?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/levelup/index.d.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
-// Type definitions for levelup 4.3
+// Type definitions for levelup 5.1
```
Comment (martin-badin) [inline types/levelup/index.d.ts:1]: The types aren't identical with 5.x.x Please see the release documentation
https://github.com/Level/levelup/releases

1. You need to remove the default export.
2. add new method `db.getMany(keys)`
3. add new read only property `supports.status`
4. and something more https://github.com/Level/levelup/commit/e1ecad9

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/levelup/tslint.json (hunk 1 above)
Comment (istherepie) [reply types/levelup/tslint.json:13]: I explained it in the comment above (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57663#issuecomment-991283057)

Do correct me if I am mistaken!

Comment (amcasey) [reply types/levelup/tslint.json:13]: Sorry, I did miss your comment.  Still, when I change this file to the preferred form (just `extends`) it seems to pass?

Regardless, as the bot [mentioned](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57663#issuecomment-991160857), it seems like the exports of this package are inconsistent with those of the underlying library?

File: types/levelup/index.d.ts (hunk 2 above)
Comment (istherepie) [reply types/levelup/index.d.ts:1]: As far as I can see `db.getMany(keys)` is the only one missing - but in either case I will make that change in a separate PR. Thanks!

File: types/levelup/tslint.json (hunk 1 above)
Comment (istherepie) [reply types/levelup/tslint.json:13]: You are correct - also according to the comment from [Martin-badin](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57663#discussion_r767104470) - the default export has been removed in version 5.0.
I will continue to bring it in line with the latest version of the library.

File: types/levelup/index.d.ts (hunk 2 above)
Comment (istherepie) [reply types/levelup/index.d.ts:1]: Actually I will continue to bring the types up to date with the current version of the library - it makes more sense rather than working around the issue.

**Review by MeirionHughes — APPROVED** [review]

**Comment by elibarzilay** [comment]
What's the status here?

**Comment by carsonfarmer** [comment]
Well, the timing is a bit rough, probably lots of folks taking time off over the past week or so. Looks like @amcasey requested specific changes, so I'll leave the final 👍 to them. With that in mind, this looks good to me. Thanks for the swift response and updates.

**Review by carsonfarmer — APPROVED** [review]

**Comment by istherepie** [comment]
> Well, the timing is a bit rough, probably lots of folks taking time off over the past week or so. Looks like @amcasey requested specific changes, so I'll leave the final 👍 to them. With that in mind, this looks good to me. Thanks for the swift response and updates.

Thanks for for your response - is anything else needed from me in this PR ?

**Review by martin-badin — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]
